src/components/NavBar/index.tsx

```diff
@@ -5,31 +5,46 @@ import Icon from "components/Icon";
 import PopoverButton from "components/PopoverButton";
 import SearchBar from "components/SearchBar";
 import { StyledLink } from "components/styles";
+import { SearchBarActions } from "constants/enums";
 import { QueryParams } from "constants/queryParams";
 import { navbarHeight, size } from "constants/tokens";
 import { useLogContext } from "context/LogContext";
 import { useQueryParam } from "hooks/useQueryParam";
 import { validateRegexp } from "utils/validators";
+import SearchResults from "./SearchResults";
 import UploadLink from "./UploadLink";
 
 const { gray, white } = palette;
 
 const NavBar: React.FC = () => {
-  const [, setSearch] = useQueryParam(QueryParams.Search, "");
   const [filters, setFilters] = useQueryParam<string[]>(
     QueryParams.Filters,
     []
   );
-  const { hasLogs, clearLogs } = useLogContext();
+  const { hasLogs, clearLogs, setSearch, searchState, paginate } =
+    useLogContext();
 
+  const { hasSearch } = searchState;
   const handleSearch = (selected: string, value: string) => {
-    if (selected === "search") {
+    if (selected === SearchBarActions.Search) {
       setSearch(value);
-    } else if (selected === "filter" && !filters.includes(value)) {
+    } else if (
+      selected === SearchBarActions.Filter &&
+      !filters.includes(value)
+    ) {
       setFilters([...filters, value]);
     }
   };
 
+  const handleOnChange = (selected: string, value: string) => {
+    if (selected === SearchBarActions.Search) {
+      setSearch(value);
+    }
+  };
+
+  const shouldClearOnSubmit = (selected: string) =>
+    selected === SearchBarActions.Filter;
```

Comment:
I feel like we don't really need to have this as a function / props. It might be better and easier to read if we just define it directly in `Searchbar` as 

```
const shouldClearOnSubmit = selected === SearchBarActions.Filter
```

I don't think it should affect the additional Jest tests you wrote for `SearchBar` very much at all